src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java

```diff
@@ -221,4 +234,29 @@ private String getReportURL(String projectUrl,String path,String projectId) {
                 .append(projectId);
         return sb.toString();
     }
+
+    @Override
+    public Cmdb getCmdb(String projectName, String version) throws HygieiaException {
+        List<CodeQuality> codeQualities = codeQualityRepository.findByNameAndVersion(projectName, version);
```

Comment:
@nameisaravind  instead of matching projectName and version from CodeQuality collection, use projectName and collectorId and match with collectorItems. Once match is found, get all components tied to matched collector-items and then find all dashboards for found components.  This way even though there is no scans on sonar project, we can still find the co-relation between sonar project and dashboard.

Reply:
@nireeshT updated as requested.